pkg/config/types/orchestrator.go

```diff
@@ -22,6 +24,18 @@ type OrchestratorAuth struct {
 	Token string `yaml:"Token,omitempty" json:"Token,omitempty"`
 }
 
+type OrchestratorNats struct {
+	// ServerTLSKey specifies the private key file path given to NATS server to serve TLS connections.
+	ServerTLSKey string `yaml:"ServerTLSKey,omitempty" json:"ServerTLSKey,omitempty"`
+	// ServerTLSCert specifies the certificate file path given to NATS server to serve TLS connections.
+	ServerTLSCert string `yaml:"ServerTLSCert,omitempty" json:"ServerTLSCert,omitempty"`
+	// ServerTLSTimeout specifies the TLS timeout, in seconds, set on the NATS server.
+	ServerTLSTimeout int `yaml:"ServerTLSTimeout,omitempty" json:"ServerTLSTimeout,omitempty"`
```

Comment:
I would prefer to keep using `Server` since it is explicitly stating the role of the cert. If we say `TLSKey` or `TLSCert` might be confusing with Mutual TLS.

Regarding the timeout, I would prefer to keep it now, but default it to a certain value, since the library default value is 2 seconds, which is short for our needs.

I will change the configuration to live in `SystemConfig`